events/processor/index.js

```diff
@@ -0,0 +1,77 @@
+import AWS from 'aws-sdk';
+import uuid from 'uuid';
+
+import Echo from './types/echo';
+
+export default class Processor {
+  constructor(event) {
+    this.kinesis = new AWS.Kinesis();
+    this.event = event;
+    this.types = {
+      'echo': Echo
+    };
+  }
+
+  process(callback) {
+    if (!this.event.hasOwnProperty('Records') ||
+        !Array.isArray(this.event.Records)) {
+      throw "event.Record is missing; this is bad!";
+    }
+    
+    this.event.Records.forEach((record) => {
+      try {
+        const data = this.getData(record);
+        const processor = new this.types[data.type](data);
+        processor.process();
+      } catch (e) {
+        this.badEvent(e, record);
+      }
+    });
+
+    if (typeof callback === 'function') {
+      callback();
+    }
+  }
+
+  getData(record) {
+    if (!record.hasOwnProperty('kinesis') || !record.hasOwnProperty('data')) {
+      throw TypeException("Malformed Kinesis record", this.record);
+    }
+
+    try {
+      const buf = new Buffer(this.record.kinesis.data, 'base64').toString('ascii');
+      const data = JSON.parse(buf);
```

Comment:
You should consider putting the decode and the parse into separate try/catch or make the error a generic error and include the error in the message. It is possible to base64 decode invalid json which then fails to parse.